items/src/main/scala/weco/api/items/services/SierraItemUpdater.scala

```diff
@@ -92,17 +97,53 @@ class SierraItemUpdater(sierraSource: SierraSource)(
 
     } yield accessConditions
 
-  private def getAvailableDates: Option[List[AvailabilitySlot]] =
-    // call Prismic to get opening times
-    // generate list of dates on now + opening times
-    Some(
-      List(
-        AvailabilitySlot(
-          "2022-01-10T10:00:00+0000",
-          "2022-01-11T10:00:00+0000"
-        )
-      )
+  private def setAvailableDates(
+    item: DisplayItem
+  ): Future[DisplayItem] = {
+    // there is only ever one location per physicalItem and one accessCondition per location,
+    // but this may sometimes be empty if it could not be fetched.
+    val accessCondition = item.locations.headOption
+      .flatMap(_.accessConditions.headOption)
+
+    if ((accessCondition.isDefined && accessCondition.head.isRequestable)) {
+      getAvailableDates(accessCondition.head)
+        .map(availableDates => item.copy(availableDates = Some(availableDates)))
+    } else {
+      Future.successful(item)
+    }
+  }
+
+  private def getAvailableDates(
+    accessCondition: DisplayAccessCondition
+  ): Future[List[AvailabilitySlot]] = {
+    val venueFromDisplayAccessCondition = Map(
+      "online-request" -> "library"
+      // other venue to be added as DisplayAccessMethod id -> content-api venue title
     )
+    val hourNow = clock.getHour
+    val leadTime = accessCondition.method.id match {
+      case "online-request" if hourNow < 10  => 1
+      case "online-request" if hourNow >= 10 => 2
+    }
+
+    venueOpeningTimesLookup
+      .byVenueName(
+        venueFromDisplayAccessCondition.get(accessCondition.method.id)
+      )
+      .map {
+        case Right(venue) =>
+          venue.openingTimes
+            .map(
+              openClose => AvailabilitySlot(openClose.open, openClose.close)
+            )
+            .drop(leadTime)
```

Comment:
```suggestion
        case Right(venue) =>
          venue.openingTimes
            .map(
              openClose => AvailabilitySlot(openClose.open, openClose.close)
            )
            .dropWhile(slot => daysAwayFromNow(slot) < leadTimeDays)
```

This suggestion assumes the existence of a new method that might look a bit like this:
```scala
def daysAwayFromNow(slot: AvailabilitySlot): Int = {
  val closeTime = ZonedDateTime.parse(slot.close)
  ChronoUnit.between(clock.timeAtVenue, closeTime, ChronoUnit.DAYS)
}
```

Reply:
Now using dropWhile, added a comment to explain why we're dropping slots too